amr-wind/utilities/sampling/VolumeSampler.cpp

```diff
@@ -0,0 +1,79 @@
+#include <limits>
+
+#include "amr-wind/utilities/sampling/VolumeSampler.H"
+
+#include "AMReX_ParmParse.H"
+
+namespace amr_wind::sampling {
+
+VolumeSampler::VolumeSampler(const CFDSim& /*unused*/) {}
+
+VolumeSampler::~VolumeSampler() = default;
+
+void VolumeSampler::initialize(const std::string& key)
+{
+    amrex::ParmParse pp(key);
+
+    pp.getarr("axis", m_axis);
+    pp.getarr("origin", m_origin);
+    pp.getarr("num_points", m_npts_dir);
+    AMREX_ALWAYS_ASSERT(static_cast<int>(m_axis.size()) == AMREX_SPACEDIM);
+    AMREX_ALWAYS_ASSERT(static_cast<int>(m_origin.size()) == AMREX_SPACEDIM);
+    AMREX_ALWAYS_ASSERT(static_cast<int>(m_npts_dir.size()) == AMREX_SPACEDIM);
+
+    // Update total number of points
+    const size_t tmp = m_npts_dir[0] * m_npts_dir[1] * m_npts_dir[2];
+    if (tmp > static_cast<size_t>(std::numeric_limits<int>::max())) {
+        amrex::Abort(
+            "VolumeSampler: Plane definition " + key +
+            " exceeds 32-bit integer limits");
+    }
+    m_npts = static_cast<int>(tmp);
+}
+
+void VolumeSampler::sampling_locations(SampleLocType& locs) const
+{
+    locs.resize(m_npts);
+
+    amrex::Array<amrex::Real, AMREX_SPACEDIM> dx;
```

Comment:
you could `const` this with declaring `dx = {m_axis[0] / m_npts_dir[0], etc};` and remove the filling of the values below

Reply:
fixed